ui/src/containers/AppDetailsContainer.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import AppsTable from '../components/AppsTable';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { Checkbox, TextContent, Text, TextVariants } from '@patternfly/react-core';
+import { getApps, reverseAppDetailsTableSort } from '../actions/actions-ui';
+import './AppDetailsContainer.css';
+
+export class AppDetailsContainer extends React.Component {
+  componentWillMount () {}
+
+  handleChange = (e) => {
+    console.log('checkbox clicked');
+  };
+
+  onSort = (_event, index) => {
+    this.props.reverseAppDetailsTableSort(index);
+  };
+
+  getTable = () => {
+    const renderedRows = [];
+    for (let i = 0; i < this.props.apps.length; i++) {
+      const tempRow = [];
+      tempRow[0] = this.props.apps[i][0];
+      tempRow[1] = this.props.apps[i][1];
+      tempRow[2] = this.props.apps[i][2];
+      tempRow[3] = this.props.apps[i][3];
+      const cb = (
+        <React.Fragment>
+          <Checkbox
+            label=""
+            isChecked={this.props.apps[i][4]}
+            onChange={this.handleChange}
+            aria-label="controlled checkbox example"
+            id={i.toString()}
+          />
+        </React.Fragment>
+      );
+      tempRow[4] = cb;
+      tempRow[5] = this.props.apps[i][5];
+      renderedRows.push(tempRow);
+    }
+
+    const caption = (
+      <TextContent>
+        <Text component={TextVariants.h4}>Deployed Versions</Text>
+      </TextContent>
+    );
+
+    return (
+      <div className="apps-table">
+        <AppsTable
+          caption={caption}
+          columns={this.props.columns}
+          rows={renderedRows}
+          sortBy={this.props.sortBy}
+          onSort={this.onSort}
+          onRowClick={this.onRowClick}
+        />
+      </div>
+    );
+  };
+
+  render () {
+    if (this.props.isAppsRequestFailed) {
+      return (
+        <div className="no-apps">
+          <p>Unable to fetch any apps</p>
+        </div>
+      );
+    }
+    return this.getTable();
+  }
+}
+
+AppDetailsContainer.propTypes = {
+  apps: PropTypes.array.isRequired,
+  sortBy: PropTypes.object.isRequired,
+  columns: PropTypes.array.isRequired,
+  isAppsRequestFailed: PropTypes.bool.isRequired
+};
+
+function mapStateToProps (state) {
+  return {
+    apps: state.appDetailRows,
+    sortBy: state.sortBy,
+    columns: state.appDetailColumns,
+    isAppsRequestFailed: state.isAppsRequestFailed
+  };
+}
+
+export default withRouter(connect(mapStateToProps, { reverseAppDetailsTableSort, getApps })(AppDetailsContainer));
```

Comment:
I think this will work without `withRouter`. It works for me but maybe there is something that we need that I am unaware of.

```jsx
export default connect(mapStateToProps, { reverseAppDetailsTableSort, getApps })(AppDetailsContainer);
```

Reply:
@craicoverflow you are correct. Unless we are doing some navigation from within the component, it's not needed here.

Reply:
good points @craicoverflow @laurafitzgerald 

@laurafitzgerald if we remove `withRouter` do we lose back button navigation?

Reply:
Going to resolve this for now, as it can be removed at a later data if we decide routing is not needed from this view

Reply:
@damienomurchu no, the back button navigation wont be lost if we remove withRouter. This is managed by the global Router object.

Reply:
Ok, removed this. Resolving now.